src/__experimental__/components/input/input.stories.js

```diff
@@ -0,0 +1,215 @@
+  const reverse = boolean('reverse (checkable inputs)', true);
+  const fieldHelpInline = disableFieldHelp ? '' : boolean('fieldHelpInline (checkable inputs)', false);
+
+  return (
+    <State store={ store }>
+      {state => <>
+        <Decimal
+          id='decimal'
+          label='Decimal Component'
+          fieldHelp={ disableFieldHelp ? '' : 'Decimal component fieldHelp' }
+          { ...commonProps }
+        />
+        <NumberInput
+          id='number'
+          label='Number Input Component'
+          onChange={ ev => store.set({ numberInputValue: ev.target.value }) }
+          fieldHelp={ disableFieldHelp ? '' : 'Number Input component fieldHelp' }
+          value={ state.numberInputValue }
+          { ...commonProps }
+        />
+        <GroupedCharacter
+          id='grouped-character'
+          label='Grouped Character Component'
+          separator='-'
+          groups={ [2, 2, 4] }
+          onChange={ ev => store.set({ groupedCharacterValue: ev.target.value.rawValue }) }
+          fieldHelp={ disableFieldHelp ? '' : 'Grouped Character Component fieldHelp' }
+          value={ state.groupedCharacterValue }
+          { ...commonProps }
+        />
+        <Select
+          ariaLabel='singleSelect'
+          label='Select Validation'
+          fieldHelp={ disableFieldHelp ? '' : 'Select Component fieldHelp' }
+          { ...commonProps }
+        >
+          <Option
+            key='option1'
+            text='Option 1'
+            value='option1'
+          />
+          <Option
+            key='option2'
+            text='Option 2'
+            value='option2'
+          />
+          <Option
+            key='option3'
+            text='Option 3'
+            value='option3'
+          />
+        </Select>
+        <Textarea
+          name='textarea'
+          label='Textarea Validation'
+          labelHelp='Returns an error when the field is empty'
+          fieldHelp={ disableFieldHelp ? '' : 'Textarea fieldHelp' }
+          { ...commonProps }
+        />
+        <Checkbox
+          id='checkbox'
+          name='my-checkbox'
+          label='Checkbox label'
+          labelHelp='Checkbox label help'
+          fieldHelp={ disableFieldHelp ? '' : 'Checkbox field help' }
+          value='checkbox-val'
+          reverse={ reverse }
+          fieldHelpInline={ fieldHelpInline }
+          onChange={ ev => store.set({ checkboxValue: ev.target.checked }) }
+          checked={ state.checkboxValue }
+          unblockValidation
+          { ...commonProps }
+          inputWidth={ undefined }
+          labelWidth={ disableLabelWidth ? undefined : commonProps.labelWidth }
+        />
+        <RadioButtonGroup
+          name='radio-button-group'
+          legend='Please select a frequency from the options below'
+          onChange={ ev => store.set({ dateValue: ev.target.value }) }
+        >
+          <RadioButton
```

Comment:
There's a lot of duplicate code for these 3 `<RadioButton>` components. I think it could be DRY'd up, eg:
```jsx
{[1, 2, 3].map(num => (
  <RadioButton
    key={`radio${num}`}
    label={`Radio ${num}`}
```

Reply:
good point, changed

Reply:
Great thanks, looks much better! :white_check_mark: